src/lib/api.mli

```diff
@@ -138,8 +139,9 @@ module type BASIC_STRING = sig
   (** [make size char] builds a new string of the passed [length] where the
       character at every position is [char], like [String.make].
 
-      @raise Invalid_argument if size is [< 0] or
-      [> {max_string_length}] if it is bounded.*)
+      [make size] may raise an exception when [size] is [< 0] or
+      [> {max_string_length}] (if it is [Some _]) depending on the
+      backend implementing the API. *)
```

Comment:
I think we're describing the api. Yes, there is no way of making sure via types that this exception is thrown, but documenting this way seems reasonable.